padguidedb/padguidedb.py

```diff
@@ -35,19 +35,15 @@ def __init__(self, bot: Red, *args, **kwargs):
         self.bot = bot
         self.settings = PadGuideDbSettings("padguidedb")
 
-        self.executor = concurrent.futures.ThreadPoolExecutor(max_workers=1)
         self.queue_size = 0
-        self.full_etl_running = False
-        self.extract_images_running = False
-
+        self.full_etl_lock = asyncio.Lock()
+        self.extract_images_lock = asyncio.Lock()
+        self.semaphore = asyncio.BoundedSemaphore(5)
         global PADGUIDEDB_COG
         PADGUIDEDB_COG = self
 
     def get_connection(self):
-        with open(self.settings.configFile(), 'a+') as f:
-            f.seek(0)
-            db_config = json.load(f)
-        return self.connect(db_config)
+        return self.connect(json.load(open(self.settings.configFile(), 'r')))
```

Comment:
I have no idea what the hell I was thinking when I wrote the code on the left.

Regardless, you should still use 'with open' instead of unguarded opens.